node/src/neighborhood/gossip_acceptor.rs

```diff
@@ -1518,57 +1563,89 @@ mod tests {
-            &RemoveStreamMsg {
-                local_addr: SocketAddr::from (db.root().node_addr_opt().unwrap()),
-                peer_addr: SocketAddr::new(old_addr.ip_addr(), old_addr.ports()[0]),
-                stream_type: RemovedStreamType::Clandestine,
-                sub: dispatcher_sub.clone()
-            }
+            &make_expected(0)
         );
         assert_eq!(
             stream_handler_pool_recording.get_record::<RemoveStreamMsg>(1),
-            &RemoveStreamMsg {
-                local_addr: SocketAddr::from (db.root().node_addr_opt().unwrap()),
-                peer_addr: SocketAddr::new(old_addr.ip_addr(), old_addr.ports()[1]),
-                stream_type: RemovedStreamType::Clandestine,
-                sub: dispatcher_sub
-            }
+            &make_expected(1)
+        );
+    }
+
+    #[test]
+    fn ipchange_accepting_connections_but_without_node_addr_is_rejected() {
+        let old_addr = NodeAddr::new(&IpAddr::from_str("1.2.3.4").unwrap(), &vec![2000, 2001]);
+        let new_addr = NodeAddr::new(&IpAddr::from_str("2.3.4.5").unwrap(), &vec![2000, 2001]);
+        let gossip_source = SocketAddr::from(new_addr.clone());
+        let (new_ipchange_node, src_db, mut dest_db) = make_ipchange_databases(
+            2345,
+            Some(old_addr.clone()),
+            Some(new_addr.clone()),
+            Mode::Standard,
+        );
+        let gossip = GossipBuilder::new(&src_db)
+            .node(new_ipchange_node.public_key(), false)
+            .build();
+        let agrs_vec: Vec<AccessibleGossipRecord> = gossip.try_into().unwrap();
+        let (stream_handler_pool, _, _) = make_recorder();
+        let (dispatcher, _, _) = make_recorder();
+        let stream_handler_pool_sub = stream_handler_pool.start().recipient();
+        let dispatcher_sub = dispatcher.start().recipient();
+        let subject =
+            IpChangeHandler::new(stream_handler_pool_sub, dispatcher_sub, Logger::new("test"));
+
+        let result = subject.qualifies(&mut dest_db, agrs_vec.as_slice(), gossip_source);
+
+        assert_eq!(
+            result,
+            Qualification::Malformed(
+                "IpChange Gossip: Node AgMEBQ accepts connections but provides no NodeAddr"
+                    .to_string()
+            ),
         );
     }
 
     #[test]
-    fn properly_constructed_originate_only_ipchange_is_identified_and_handled() {
-        let old_addr = NodeAddr::new (&IpAddr::from_str("1.2.3.4").unwrap(), &vec![2000, 2001]);
-        let new_addr = NodeAddr::new (&IpAddr::from_str("2.3.4.5").unwrap(), &vec![2000, 2001]);
-        let gossip_source = SocketAddr::from (new_addr.clone());
-        let (gossip, new_node, mut db) =
-            make_ipchange (2345, Some (old_addr.clone()), Some (new_addr),
-                           Mode::OriginateOnly);
+    fn properly_constructed_originate_only_ipchange_is_identified_and_ignored() {
```

Comment:
Actually. I've got another question, after thinking about originate only Nodes again. 

Why is that we cannot reestablish a new connection with them? I understand that **other** Nodes cannot initiate the process. However, if we look at the standard debut Gossip, I suppose any originate only Node must make initiative and send it and later accept fully, if the other party also decided to accept. 
I also know the IpChange Gossip is extremely similar to a Debut. 

So why is it that we cannot handle this reconstitution of the relationship (perhaps shadowing the procedure happening during handling the Debut etc., in other words, the originate only one is the active one in this so why we cannot absorb it and instead the ignored part has to be played?)

Reply:
Good...I think we should keep it.